code/modules/lighting/lighting_atom.dm

```diff
@@ -2,21 +2,25 @@
 // The proc you should always use to set the light of this atom.
 // Nonesensical value for l_color default, so we can detect if it gets set to null.
 #define NONSENSICAL_VALUE -99999
-/atom/proc/set_light(l_range, l_power, l_color = NONSENSICAL_VALUE)
+/atom/proc/set_light(l_range, l_power, l_color = NONSENSICAL_VALUE, l_on)
 	if(l_range > 0 && l_range < MINIMUM_USEFUL_LIGHT_RANGE)
 		l_range = MINIMUM_USEFUL_LIGHT_RANGE	//Brings the range up to 1.4, which is just barely brighter than the soft lighting that surrounds players.
-	if (l_power != null)
+	if (!isnull(l_power))
 		light_power = l_power
 
-	if (l_range != null)
+	if (!isnull(l_range))
 		light_range = l_range
 
 	if (l_color != NONSENSICAL_VALUE)
 		light_color = l_color
 
-	SEND_SIGNAL(src, COMSIG_ATOM_SET_LIGHT, l_range, l_power, l_color)
+	if(!isnull(l_on))
+		light_on = l_on
 
-	update_light()
+	SEND_SIGNAL(src, COMSIG_ATOM_SET_LIGHT, l_range, l_power, l_color, l_on)
+
+	if(light_system == STATIC_LIGHT)
```

Comment:
Maybe add this check at the beginning of the proc, crashing then?
We want these values to be changed by the appropriate procs if using a different lighting system.

Reply:
> 
> 
> > However, I found out a runtime. set_light() is often being called on some dynamic lightings, and then calls update_light(), which prints a CRASH. I'll make it only update when it's `light_system == STATIC_LIGHT`
> 
> This sounds like something we could instead cut at the source, keeping the informative `CRASH` message, no?

Oh my bad, I thought the dynamic lighting was subscribed to COMSIG_ATOM_SET_LIGHT signal, but it turns out it isn't. I'll revert that change

Reply:
oh what where did I comment

Reply:
the now called update_light() will CRASH it